argo-cd-apps/base/host/segment-bridge/segment-bridge-host.yaml

```diff
@@ -12,9 +12,11 @@ spec:
               values:
                 sourceRoot: components/segment-bridge
                 environment: staging
-                clusterDir: ""
+                clusterDir: base
           - list:
-              elements: []
+              elements:
+                - nameNormalized: stone-stg-host
+                  values.clusterDir: stone-stg-host
```

Comment:
Intresting, but won't it use all the directories under `staging`? meaning that the settings for the host cluster will include both`stone-stg-host` and `stone-stg-rh01` and `stone-sth-m01`, instead of the current setting where I can have it use `stone-stg-host` only for the host configurations, and the others for the member configurations?

Reply:
no it will not. This generators spec determine how to create application. The `clusters` generator will create one application per cluster which matches the criteria. You do not see the criteria in this file because it is added by another kustomization file. If you do the kustomize build, you will see the criteria:
```
hares@hares-thinkpadp1gen3 infra-deployments]$ kustomize build argo-cd-apps/overlays/staging/ | grep segment-bridge-host$ -A29
  name: segment-bridge-host
  namespace: argocd-staging
spec:
  generators:
  - merge:
      generators:
      - clusters:
          selector:
            matchLabels:
              appstudio.redhat.com/host-cluster: "true"
          values:
            clusterDir: base
            environment: staging
            sourceRoot: components/segment-bridge
      - list:
          elements:
          - nameNormalized: stone-stg-host
            values.clusterDir: stone-stg-host
      mergeKeys:
      - nameNormalized
```
The `clusters` will only create one application per cluster having `appstudio.redhat.com/host-cluster: "true"` label.

That said, maybe we should proceed with the change the way it is now as it is consistent with other components and refactoring could be done separately, for all components at once.

Reply:
I see, OK. 
It's a very easy change, but since you already mentioned it - it's probably wise to wait for a repo-wide refactor for consistency.